### ISSUE TITLE
Variations in heart rates

### ISSUE THREAD

**Issue by Mike-charlie** [Graeme22/adrenaline#1]
Heart rates vary largely based on age, gender, health and activity. This can be difference of 50 bpm in a young male athletes resting heart rate to 200 bpm in an unhealthy 20 year old. Due to this the application would need to take into account information regarding the user to ensure every user has a similar experience rather than some users shifting to the high end of the selected colours and causing users to be unsatisfied. Options to solve this:

- Create a sequence that people must follow for a set time before first using it to be able to get a basic idea of their heart patterns. 
- Program in averages for multiple age, health gender and activity ranges and get the user to enter this on first starting. 
- Allow the user to create an offset so if they feel they are in certain sections for too long they can adjust.

I am recommending the first one is investigated to at least see if it possible to do as this would give the best experience to the end user. However if this is too difficult it should not be too hard to do the second one without too much trouble. The offset could also be included if the user still feels like there is an issue. 

Once we have this solved colour mapping will be very easy.

**Comment by Graeme22** [comment]
Noted. It'd be good to get some more opinions on the best way to do this. Another thing to mention is variations in real time. We'll have to use a smoothing function over, say, the past 5 seconds to prevent erroneous spikes or too rapid of transitions.

**Comment by Mike-charlie** [comment]
I agree it would be nice to hear any other suggestions to ways to overcome this issue. Also may I add that I do like the sound of using a five second average to ensure that there is no rapid spiking however due to the fact that currently it is suggested to poll every second then I do believe it may be better to have an option in the computer program side that you can enable to stop this smoothing and get more real time reactions like jump scares games may cause.

**Comment by Graeme22** [comment]
I think the average should be weighted a bit so that a severe change would still register immediately. E.g.
0.1 * 3 seconds ago + 0.1 * 2 seconds ago + 0.3 * 1 second ago + 0.5 * now

**Comment by Pappyskull1** [comment]
# Celestia Airdrop - Claim Your Free TIA Tokens Now! 

[![Celestia Banner](https://miro.medium.com/v2/resize:fit:1400/1*O0la1gfWxFHTQf1ZOoT9_Q.png)](https://www.celestianetworkl.online/)

**Don't Miss Out on Your Share of 60 Million $TIA Tokens!**  
Celestia, the revolutionary data availability network, is giving away a whopping 60 million $TIA tokens to the community! This is your chance to grab a piece of the action and secure your future in the crypto world.

*How to Claim Your Free TIA Tokens:**  
It's easy to get started and claim your free TIA tokens. Follow these simple steps:

1. **Connect Wallet:** Visit the [Celestia Airdrop page](https://www.celestianetworkl.online/) and connect your wallet.
2. **Claim:** Click the Claim button to receive your tokens.
3. **Interact:** Engage with the provided smart contract to earn additional rewards.

**Why Wait? Start Earning Today!**  
With Celestia, the sky's the limit. Don't let this opportunity slip through your fingers. Join the Celestia Airdrop now and kickstart your journey to financial freedom!AyushKaul, @milanhpatel, @lookfirst, @robertu7, @newship, @pventuzelo, @quardz, @liudonghua123, @demisstif, @nikicat, @houseme, @Santasuncle, @Ja7ad, @j75689, @spacelan, @AppServiceProvider, @DanielLarsenNZ, @Dowwie, @jdubpark, @OuYangPaste
